certbot/certbot/_internal/main.py

```diff
@@ -726,10 +727,14 @@ def _tos_cb(terms_of_service: str) -> None:
                 display_util.notify("Account registered.")
             except errors.MissingCommandlineFlag:
                 raise
-            except errors.Error:
+            except (errors.Error, acme_messages.Error) as err:
                 logger.debug("", exc_info=True)
+                if isinstance(err, acme_messages.Error):
+                    err_msg = f"Error returned by the ACME server: {str(err)}"
```

Comment:
Lol:

https://github.com/certbot/certbot/blob/142fcad28b0e61f335b92a743b38c8d677bd00dd/certbot/certbot/_internal/log.py#L367-L370

I am starting to wonder we should change this properly, now. I'm actually not sure whether messing with `__str__` would be a breaking change, but we can always add a new method for display purposes. Let me find out.

Reply:
Hm, that code *assumes* that an ACME error *always* contains all three properties. But in `acme.messages.Error` all three properties have `omitempty=True` set. So if the ACME server doesn't e.g. provide the title for some reason, `str(err)` would only provide two partitions separated by `:: `, right? Not sure if Boulder actually does that and not sure if it's allowed by the RFCs, but still.

Also, not sure how it's tied into this PR.

Reply:
I would really like to not show the error type and the description field, but let's leave it for now. This PR fixes the problem it was supposed to, albeit slightly verbosely.